JOSS paper

navidcy opened the pull request:
Added the JOSS template. Let's work on this! 🎉

After every commit that modifies the `paper.md` markdown file a GitHub action is called that compiles the PDF version of the paper and commits it in the `paper` directory. See, e.g., [9ea312f](https://github.com/COSIMA/regional-mom6/pull/100/commits/9ea312fa76f2c31b292f0c380e810688004a7e09).

navidcy commented:
Relevant discussion in #42

ashjbarnes commented:
Is the workflow here to keep this PR open as we work on the paper together and merge when we're happy?

navidcy commented:
Yes, that's the idea. Keep the PR open, work on it, submit to JOSS (still the PR is open) and we merge only when it's accepted after the reviews etc.

navidcy commented on paper/paper.md:
it's convenient if you have the habit of starting a new line for every sentence.
this way will make commenting on the PR easier.... Otherwise 1 line is a whole paragraph so it's very hard for somebody to suggest a "code change" to fix a typo.

navidcy commented on paper/paper.md:
```diff
@@ -0,0 +1,77 @@
+
+# Summary
+
+Modular Ocean Model 6 (MOM6) is a widely used general circulation ocean model developed at the Geophysical Fluid Dynamics Laboratory (GFDL) at Princeton. Among other improvements on its predecessor MOM5, this iteration permits open boundary conditions, and MOM6 is subsequently growing in popularity for high resolution regional modelling. However, setting up a regional domain can be challenging an ...
```
need citation for MOM6; we should cite:

```bibtex
@article{Adcroft2019MOM6,
    author = {Adcroft, Alistair and Anderson, Whit and Balaji, V. and Blanton, Chris and Bushuk, Mitchell and Dufour, Carolina O. and Dunne, John P. and Griffies, Stephen M. and Hallberg, Robert and Harrison, Matthew J. and Held, Isaac M. and Jansen, Malte F. and John, Jasmin G. and Krasting, John P. and Langenhorst, Amy R. and Legg, Sonya and Liang, Zhi and McHugh, Colleen and Radhakrishnan, Aparna and Reichl, Brandon G. and Rosati, Tony and Samuels, Bonita L. and Shao, Andrew and Stouffer, Ronald and Winton, Michael and Wittenberg, Andrew T. and Xiang, Baoqiang and Zadeh, Niki and Zhang, Rong},
    title = {The {GFDL} global ocean and sea ice model {OM4.0}: {Model} description and simulation features},
    journal = {Journal of Advances in Modeling Earth Systems},
    volume = {11},
    number = {10},
    pages = {3167-3211},
    doi = {10.1029/2019MS001726},
    year = {2019}
}
```

navidcy commented on paper/paper.md:
```diff
@@ -0,0 +1,77 @@
+
+# Statement of need
+
+The learning curve for setting up a regional ocean model can be quite steep. In the case of MOM6, there are several tools scattered around github like those collected in [ESMG's grid tools](https://github.com/ESMG/gridtools), as well as examples hardcoded for particular domains, input files and hardware. However, there is no one-stop-shop to learn how to get a regional MOM6 model up and running,  ...
```
we should properly cite Github repositories; see example in the `.bib` file, i.e.

```bibtex
@misc{fidgit,
  author = {A. M. Smith and K. Thaney and M. Hahnel},
  title = {Fidgit: An ungodly union of GitHub and Figshare},
  year = {2020},
  publisher = {GitHub},
  journal = {GitHub repository},
  url = {https://github.com/arfon/fidgit}
}
```

navidcy replied:
if there are a bunch of contributors you can add as authors the first 2-3 and then `and contributors},`

ashjbarnes replied:
Thanks yes proper referencing was on my todo later I was just trying to brain dump for now!

navidcy replied:
brain dump away!

navidcy commented on paper/paper.md:
```diff
@@ -0,0 +1,77 @@
+
+# Acknowledgements
+
+We acknowledge contributions from Brigitta Sipocz, Syrtis Major, and Semyeong
+Oh, and support from Kathryn Johnston during the genesis of this project.
```
```suggestion
We thank the Consortium for Ocean–Sea Ice Modeling in Australia ([cosima.org.au](https://cosima.org.au)) for useful discussions during the development of this package.
N.C.C. acknowledges funding from the Australian Research Council under DECRA Fellowship DE210100749.
```

navidcy commented on paper/paper.md:
```diff
@@ -0,0 +1,113 @@
+
+# Summary
+
+Modular Ocean Model 6 (MOM6) is a widely used general circulation ocean model developed at the Geophysical Fluid Dynamics Laboratory (GFDL) at Princeton.
```
Let's not mention "Princeton". It's not like it's the most famous city in the world... If you insist let's say "Princeton, NJ at the USA". But what's the point?

```suggestion
Modular Ocean Model 6 (MOM6) is a widely used general circulation ocean model developed at the Geophysical Fluid Dynamics Laboratory (GFDL) [@Adcroft2019MOM6].
```

navidcy commented on paper/paper.md:
```diff
@@ -0,0 +1,113 @@
+The `regional_mom6` python package automates much of the regridding, metadata encoding, grid generation and other miscellaneous steps, allowing models to be up and running more quickly.
+
+The `regional_mom6` package takes raw files containing the initial condition, forcing and bathymetry
+These inputs can be on the Arakawa A,B or C grids, and the package performs the appropriate interpolation using `xESMF` `@zhang:2020` onto the C grid required by MOM6.
```
```suggestion
These inputs can be on the Arakawa A,B or C grids, and the package performs the appropriate interpolation using `xESMF` [@zhang:2020] onto the C grid required by MOM6.
```

navidcy commented on paper/paper.md:
```diff
@@ -0,0 +1,113 @@
+The learning curve for setting up a regional ocean model can be quite steep.
+In the case of MOM6, there are several tools scattered around github like those collected in ESMG's grid tools `@gridtools`, as well as examples hardcoded for particular domains, input files and hardware.
+However, there is no one-stop-shop to learn how to get a regional MOM6 model up and running, meaning that a newcomer must collect many disparate pieces of information from around the internet unless they are able to get help.
+Other models have packages to aid in domain setup like `Pyroms` `@pyroms` for ROMS and `MITGCM_python` `@mitgcmpy` for MITGCM.
```
```suggestion
Other models have packages to aid in domain setup like `Pyroms` `@pyroms` for ROMS and `MITgcm_python` [@mitgcmpy] for the Massachusetts Institute of Technology General Circulation Model (MITgcm; [@ marshall1997finite]).
```

navidcy commented on paper/paper.md:
```diff
@@ -0,0 +1,113 @@
+However, setting up a regional domain can be challenging and time consuming for new users even for the simplest rectangular domains.
+The `regional_mom6` python package automates much of the regridding, metadata encoding, grid generation and other miscellaneous steps, allowing models to be up and running more quickly.
+
+The `regional_mom6` package takes raw files containing the initial condition, forcing and bathymetry
```
```suggestion
The `regional_mom6` package takes raw files containing the initial condition, forcing, and bathymetry
```

navidcy commented on paper/paper.md:
```diff
@@ -0,0 +1,113 @@
+
+The `regional_mom6` package takes raw files containing the initial condition, forcing and bathymetry
+These inputs can be on the Arakawa A,B or C grids, and the package performs the appropriate interpolation using `xESMF` `@zhang:2020` onto the C grid required by MOM6.
+This base grid can either be constucted based on the user's desired resolution and choice of pre-configured options, or the user can provide their own horizontal or vertical grids.
```
```suggestion
This base grid can either be constructed based on the user's desired resolution and choice of pre-configured options, or the user can provide their own horizontal or vertical grids.
```

navidcy commented on paper/paper.md:
```diff
@@ -0,0 +1,113 @@
+The package also comes with pre-configured run directories, which can be automatically copied and modified to match the user's experiment
+Subsequently, a user need only copy a demo notebook, modify the longitude, latitude and resolution, and simply by running the notebook from start to finish will generate all they need for running a MOM6 experiment in their domain of interest.
+
+This package is targeted at users with basic Python skills, and contains a documented tutorial based on publicly available forcing and bathymetry datasets. 
```
```suggestion
This package is targeted at users with basic Python skills, and contains a documented tutorial that uses a publicly available forcing and bathymetry datasets [mention which datasets, perhaps cite?].
```

ashjbarnes replied:
Yes was thinking about citing. How would you in-text cite the copernicus data? Their info page doesn't come with an example bibtex entry just says to list the DOIs of the products you used: https://help.marine.copernicus.eu/en/articles/4444611-how-to-cite-copernicus-marine-products-and-services

navidcy commented on paper/paper.md:
```diff
@@ -0,0 +1,96 @@
+
+# Summary
+
+Modular Ocean Model 6 (MOM6) is a widely used general circulation ocean model developed at the Geophysical Fluid Dynamics Laboratory (GFDL) at Princeton `[@Adcroft2019MOM6]`.
```
I don’t think you need to enclose citations in backticks.

```suggestion
Modular Ocean Model 6 (MOM6) is a widely used general circulation ocean model developed at the Geophysical Fluid Dynamics Laboratory (GFDL) at Princeton [@Adcroft2019MOM6].
```

ashjbarnes replied:
thanks yeah I found their example confusing as they did use these quotations which didn't seem to work properly

navidcy replied:
I think they were thinking you’d read it from the PDF :)

ashjbarnes commented:
Is there a good way to build / test locally? I keep pushing when the formatting breaks to test which seems inefficient!

navidcy commented:
> Is there a good way to build / test locally? I keep pushing when the formatting breaks to test which seems inefficient!

Ignore the tests anyway. We are not changing any part of the code. 

I don’t know the answer to your question. That’s how I was doing it other times. You can look into this in JOSS website but if it’s just so the “the tests don’t stop” I wouldn’t bother.

ashjbarnes commented:
Order of authors wasn't meant to be a decision too btw I was just putting them on the paper

ashjbarnes commented:
@dhruvbhagtani @ChrisC28 @reillyja @aekiss all contributors to the package on Github are welcome to contribute to the paper. Here's what we've got so far - any feedback or suggestions would be great :) Apparently a 2-3 page written report is normal for JOSS papers. We also need to make sure that the documentation / readthedocs page are in a good state for the reviewers too

navidcy commented:
@ashjbarnes could I make some edits and commit (instead of suggestions in the PR)? are you done for the moment?

navidcy commented:
The caption of the figure insists not appearing... hm...

dhruvbhagtani commented:
The manuscript looks great! Suggesting some areas of discussion (please feel free to ignore any or all of them):

1. It might be useful to have a figure explaining regional modelling like - downscaling from global to your region of interest that we can highlight (which can be Tasmania and we can then explain our example notebook as well, more on that below). 
2. I don't know if there are plans on this or not, but reading other papers in JOSS, I realised that having a more extensive example within the paper could be useful. We can use our Tassie example for this.

I can go through these example codes later in the week and make edits if needed.

navidcy commented:
I admit I haven't read the manuscript in detail yet. I'll do it today (not holiday in Cyprus) and will add comments or restructure a bit.

navidcy commented:
I also think that we should definitely mention https://gmd.copernicus.org/articles/16/1481/2023/

```bibtex
@article{polton2023reproducible,
    author  = {Polton, J. and Harle, J. and Holt, J. and Katavouta, A. and Partridge, D. and Jardine, J. and Wakelin, S. and Rulent, J. and Wise, A. and Hutchinson, K. and Byrne, D. and Bruciaferri, D. and O'Dea, E. and De Dominicis, M. and Mathiot, P. and Coward, A. and Yool, A. and Palmi\'eri, J. and Lessin, G. and Mayorga-Adame, C. G. and Le Guennec, V. and Arnold, A. and Rousset, C.},
    title   = {Reproducible and relocatable regional ocean modelling: fundamentals and practices},
    journal = {Geoscientific Model Development},
    volume  = {16},
    year    = {2023},
    number  = {5},
    pages   = {1481--1510},
    doi     = {10.5194/gmd-16-1481-2023}
}
```

navidcy commented:
> Is there a good way to build / test locally? I keep pushing when the formatting breaks to test which seems inefficient!

for future reference:
https://github.com/COSIMA/regional-mom6/blob/c51e90f712251ed5decd569d6537d2f586a39405/.github/workflows/testing.yml#L3-L7

navidcy commented on paper/paper.md:
```diff
@@ -0,0 +1,99 @@
+Modular Ocean Model 6 (MOM6) is a widely-used general circulation open-source ocean model developed mainly at the Geophysical Fluid Dynamics Laboratory (GFDL) [@Adcroft2019MOM6].
+Among other improvements on its predecessor MOM5, this iteration permits open boundary conditions, and MOM6 is subsequently growing in popularity for high resolution regional modelling.
+However, setting up a regional domain can be challenging and time consuming for new users even for the simplest rectangular domains.
+The `regional_mom6` python package automates much of the regridding, metadata encoding, grid generation and other miscellaneous steps, allowing models to be up and running more quickly.
```
@ashjbarnes, why we say "automates much of the regridding" and not just "automates ~much of~ the regridding"?
is this intentional?

ashjbarnes replied:
True - I guess I was being modest about it because it will only automate things that are currently supported. There are many use cases where you use the pipeline we've set up but customise things as you go (like the grid). The "much" was meant to be a qualifier for all of the things listed not just the regridding. We can remove it though

navidcy replied:
Let's remove it because it gives that impression "this package will do mostly the regrinding and then you would need to finish off some of the regrinding yourselves"

navidcy commented:
@ashjbarnes

Regarding the Figure:

- I don't know why the caption doesn't appear but I'll try to sort it out.
- Could we plot a more oceanic quantity? Like surface vorticity? Something that doesn't involve the forcing dataset.
- Could you please add a colorbar with units
- Don't put a title on the figure, like the "Curl of wind stress" that appears now.
- Mention in the caption whether this is a snapshot, a day average, ...
- Mention in the caption a bit more for the configuration? E.g., "from a regional configuration of the Tasman sea modeled at 1/80th degree resolution with 124 vertical levels. The configuration of this regional simulation can be found at the repository ...."?

navidcy commented:
I've gone through the manuscript one and made a few changes.

I removed the part where the cosima cookbook was mentioned -- it felt a bit out of place. I also trimmed down several details that I thought they were minor, e.g., that the package can handled the seam in the grids.

After @dhruvbhagtani's comment I think 1-2 sentence for what is regional modelling and why it's useful are appropriate. (I don't think we need a schematic to explain though.) For example something saying that by focusing on a region one can increase the resolution of the simulation and incorporate more physics in the model. And why is this useful etc.... @ashjbarnes wanna have a go writing 1-2 sentences somewhere at the beginning?

navidcy commented on paper/paper.md:
```diff
@@ -0,0 +1,112 @@
+
+![The curl of the wind stress from a regional MOM6 simulation in the Tasman sea. The source code can be found at https://github.com/ashjbarnes/tasman-tides. \label{fig:example}](tasman-speed.png){ width=80% }.
+
+Figure \autoref{fig:example} is an example of a domain set up with the `regional_mom6` package. In this case, the forcing datasets are GLORYS and ERA5, with an 80th degree horizontal resolution. 
```
Can you fill in the details I suggest below?

```suggestion
Figure \autoref{fig:example} is an example of a domain set up with the `regional_mom6` package. In this case, the forcing datasets are GLORYS and ERA5, with an 1/80-th degree horizontal resolution (around X kilometers) and XXX vertical levels. 
```

navidcy commented:
> I also think that we should definitely mention https://gmd.copernicus.org/articles/16/1481/2023/
> 
> ```
> @article{polton2023reproducible,
>     author  = {Polton, J. and Harle, J. and Holt, J. and Katavouta, A. and Partridge, D. and Jardine, J. and Wakelin, S. and Rulent, J. and Wise, A. and Hutchinson, K. and Byrne, D. and Bruciaferri, D. and O'Dea, E. and De Dominicis, M. and Mathiot, P. and Coward, A. and Yool, A. and Palmi\'eri, J. and Lessin, G. and Mayorga-Adame, C. G. and Le Guennec, V. and Arnold, A. and Rousset, C.},
>     title   = {Reproducible and relocatable regional ocean modelling: fundamentals and practices},
>     journal = {Geoscientific Model Development},
>     volume  = {16},
>     year    = {2023},
>     number  = {5},
>     pages   = {1481--1510},
>     doi     = {10.5194/gmd-16-1481-2023}
> }
> ```

@ashjbarnes have you read this paper above? where is a good place to discuss about it? where we talk about other packages for other models?

navidcy commented on paper/paper.md:
```diff
@@ -72,7 +72,7 @@ However, the package's code design and modularity allows makes more complex conf
 <!-- As more advanced use cases emerge, users can contribute their grid generation functions as well as example configuration files and notebooks.  -->
 
-![A snapshot of surface speed in MOM6 simulation in the Tasman sea. The source code can be found at https://github.com/ashjbarnes/tasman-tides. \label{fig:example}](tasman_speed.png){ width=80% }.
+![A figure](tasman_speed.png){ width=80% }.
```
this `.` at the end was the culprit for the caption not appearing!!!!!!


!!!!!!!!!!!!!!!!

navidcy replied:
cc @ashjbarnes

navidcy commented:
Suggestion for figure:

we have one global panel of the speed of ERA5 and mark the region of the regional simulation and then the other panel the one we have now?

https://cosima-recipes.readthedocs.io/en/latest/Tutorials/Making_Maps_with_Cartopy.html#Mark-a-region

navidcy commented:
moving the discussion here:

on #143 @ChrisC28 asks:

> How removed from the NCI/Gadi environment do we need to be to make this work open for a general audience?

and me and @ashjbarnes replied:

by @navidcy:
> We need to be totally removed! A package cannot be only applicable to particular machine. Imagine the [Nature paper](https://www.nature.com/articles/s41586-020-2649-2) titled "Array programming with NumPy on Australia's Gadi HPC". I doubt it would have the same audience...

> Why you ask @ChrisC28 -- did you have anything in mind? As it stands now the package _is completely removed_ from NCI-specifics. Only one of the two examples is NCI-specific, the one that uses ACCESS-OM2 output as forcing. But our plan is to remove this example from the package's documentation and move it to the cosima-recipes (see https://github.com/COSIMA/cosima-recipes/issues/326) and then add a link to the cosima-recipes in this package's docs.

 by @ashjbarnes:
> At this stage it's pretty devoid of NCI specific stuff (except for the access om2 example which we're still trying to migrate to cosima examples). One payu-centric thing is still present (#93)

ChrisC28 commented:
HI @navidcy and @ashjbarnes 

I'm happy with your collective responses. The comparisons between MOM5 and MOM6 are best left out due to the scope of the package. 

Note that I included my point about de-NCI-ing the shoftware for two reasons: 
- Experience trying to port packages developed on one piece of harward to another, and finding occasional issues where the original hardware has been baked in to the development;
- Trying to stay a step ahead of reviewers, who are likely to try to run the package on their own system, which is unlikely to be NCI. 
As such, I'm not pointing to anything specific, just trying to flag a _potential_ issue before it comes up.

ChrisC28 reviewed (approved):
Looks good to me at this point.

navidcy commented:
Great! Well I totally agree. The package needs to be machine-agnostic. And it is at this point!

navidcy commented:
Some of the above-mentioned commented parts include:

> A package also provides a standardised way of setting up regional models, allowing for more efficient troubleshooting. This is particularly important as the MOM6 boundary code is still under active development, meaning that an old example found Github may not work as intended with a newer executable. Currently, it is difficult to discern what the best model settings are for a particular experiment with a given MOM6 executable. However, having different releases of a python package tied to releases of the MOM6 executable will help users avoid difficult to diagnose compatibility errors between the MOM6 codebase, input file formats and parameter files.

I find this part a bit pointless? I think it's common knowledge that having packages with proper versioning is how one can construct a workflow that is reproducible. Are we trying to say something more than this here and I'm missing it? (ping @ashjbarnes, I believe you wrote this?)

ashjbarnes commented:
This link though needs to be better incorporated into the documentation since it's nested inside a repo that's designed to work with NCI. Currently I link it [here](https://github.com/COSIMA/regional-mom6/tree/joss-paper#mom6-configuration-and-version-requirements) in the readme

navidcy commented:
> > Some of the above-mentioned commented parts include:
> > > A package also provides a standardised way of setting up regional models, allowing for more efficient troubleshooting. This is particularly important as the MOM6 boundary code is still under active development, meaning that an old example found Github may not work as intended with a newer executable. Currently, it is difficult to discern what the best model settings are for a particular experiment with a given MOM6 executable. However, having different releases of a python package tied to releases of the MOM6 executable will help users avoid difficult to diagnose compatibility errors between the MOM6 codebase, input file formats and parameter files.
> > 
> > 
> > I find this part a bit pointless? I think it's common knowledge that having packages with proper versioning is how one can construct a workflow that is reproducible. Are we trying to say something more than this here and I'm missing it? (ping @ashjbarnes, I believe you wrote this?)
> 
> My point here is that the MOM6 code, as well as the coupler, and 4 other necessary components (I'll spare you the details) march onwards on different release schedules. As these various components have changed, some parts of the MOM6 regional setup also change but this isn't documented. I had to trawl through the Fortran code debugging things to get the format right. Whilst it's obvious I thought it was worth pointing out that we have gone to the effort of getting things working on the latest set of executables, and [bookmarked these releases so users know which ones to compile](https://github.com/ashjbarnes/setup-mom6-nci/tree/main/src).

OK -- now I understand!

> MOM6 boundary code
I though this was referring to `regional_mom6` package!

Let me put that back in now that I understand where it's coming from and you can tell me if you agree with what I write.

You brought bad dreams back in my memory! I now remembered that MOM6 do not tag their releases, nor they announce when they make breaking API changes! It's all _just happens_, and if you create an executable the next day *things could be silently failing*.... oh dear....

navidcy commented:
OK, people, I would like to have a final push on this and let the reviewers take it.
How about we aim to submit on **Tue April 23**?

I'd like to have #149 and #148 merged before we submit.
Ideally also #147 + releasing v0.5.0 since #147 include a breaking API change. But we won't let #147 hold submission back.

cc @ashjbarnes, @angus-g, @aekiss, @ChrisC28, @reillyja, @dhruvbhagtani

Just saying, that since we've heard from all of you at this point, silence will be interpret as: "*Cool! I'm really happy with the manuscript and please submit! I've also added any acknowledgments I wanted to include!*"

navidcy commented:
Note, btw, that you feedback on the README and Documentation are very useful. The reviewers won't review just the 2-3 page manuscript. They'll review the repository!

navidcy commented:
@ChrisC28 does [this section](https://regional-mom6.readthedocs.io/en/latest/mom6-file-structure-primer.html) reads OK and addresses the comment you made in https://github.com/COSIMA/regional-mom6/pull/143? There is extensive info on what are the inputs required in every docstring, e.g., see https://regional-mom6.readthedocs.io/en/latest/api.html

navidcy commented:
@ashjbarnes et al, I wrote a small bit justifying why regional ocean modeling is needed. Please have a look and modify/suggest changes accordingly. 

@ashjbarnes I also cited figure 1 as if it has the 2-panel structure we discussed: panel a) a global ocean surface currents from GLORYs and panel b) the current figure.

navidcy commented:
There is also some note in the text `[cite 1-2 repos?]` and I think it's you @ashjbarnes who wrote the statement that precedes that note. Do you have 1-2 repos in mind?

ChrisC28 commented:
THanks for your efforts @navidcy and @ashjbarnes .

I think the MOM6 file structure primer is extremely useful, not just for users of the regional configuration but for MOM6 users more generally. I think it reads well and addreses my point about the inputs and outputs to be expected by a user.

navidcy commented:
> I think the MOM6 file structure primer is extremely useful, not just for users of the regional configuration but for MOM6 users more generally. 

I agree an I think we should push to put these in the MOM6 docs.

ashjbarnes commented on paper/paper.md:
```diff
@@ -0,0 +1,122 @@
+The Modular Ocean Model version 6 (MOM6) is a widely-used open-source general circulation ocean--sea ice model, developed mainly at the NOAA Geophysical Fluid Dynamics Laboratory [@Adcroft2019MOM6].
+MOM6 contains several improvements over its predecessor MOM5 [@griffies2014elements], including the implementation of the Arbitrary-Lagrangian-Eulerian vertical coordinates [@griffies2020ALE], more efficient tracer advection schemes, and state-of-the art parameterizations of sub-grid scale physics.
+
+The nature of turbulent flows is such that smaller scales of motion emerge emerge spontaneously.
```
```suggestion
The nature of turbulent flows is such that smaller scales of motion emerge spontaneously.
```

navidcy commented:
@ashjbarnes suggested that instead of my figure suggestion we show this:

![regional](https://github.com/COSIMA/regional-mom6/assets/7112768/a308b606-055e-482a-abdb-343785305a7e)

This showcases how a regional model matches the flow at its boundaries and how we can use regional modeling to resolve more and more scales of motion.

@reillyja, you made this figure, right? is it easy to reproduce it with a few tweaks and use it in the paper? could you point me to a notebook that produces this figure perhaps?

dhruvbhagtani commented:
Looks great! I may have missed this, but as of now, the third paragraph talks about Fig (a) and (b), but I don't see any subfigures. Are we going to add something? I'll push minor corrections soon

navidcy commented:
> Looks great!
The manuscript or the package in general?

> I may have missed this, but as of now, the third paragraph talks about Fig (a) and (b), but I don't see any subfigures. Are we going to add something?

Yes. We are gonna update the figure and change the text accordingly.

dhruvbhagtani commented:
Tha manuscript is looking great! I see lots of things have changed since I last read, and I am particularly happy with how the motivation is set now.

dhruvbhagtani commented:
In paragraph 2, when we are talking about the energy cascade, will it be useful to add a citation? Perhaps one of the few already cited later in the paragraph?

navidcy commented:
Yes, I was thinking of that. What's a good reference for the statement that turbulent flows spontaneously create smaller scales of motion? The ones from the Ocean Mixing book are not that relevant here... Citing Kolmogorov 1941 or a classical Navier-Stokes paper I was a bit hesitant.

Perhaps

> Richardson L. (1922). Weather prediction by numerical process (Cambridge, England: Cambridge University Press). doi: 10.1017/CBO9780511618291

with the famous "*Big whirls have little whirls that feed on their velocity, and little whirls have lesser whirls and so on to viscosity*"?

ChrisC28 commented:
Hi all, 
Devil's advocate here: do we need to explain forward/inverse cascades in this paper? The reasons for running regional and high resolution models is diverse (improved representation of coastlines or islands or whatever, focus on a particular region too small to be resolved in coarse resolution models, etc...) and extends beyond simply including smaller scale physics. Additionally, I think this discussion is handled well by some of the papers we cite. 

Personally, I'd write something along the lines of: "regional ocean models can be run at higher resoltuions while limiting the required computational resources. Higher resolution regional ocean models can then improve the representation of smaller scale motions, such as tidal beams, mixing, meso an sub-meoscale circulation, as well the oceanic response to smaller scale bathymetruc or coastal features (such as headlands, islands, sea-mounts, or submarine canyons) and forcing (such as atmospheric fronts and convective storms). They also allow for local regions of interest to downstream users that may be contained witin a single grid cell of a coarse resolution model to be resolved.

navidcy commented:
this is much better way to motivate -- thanks!
I'll include something along those lines and ping you to have a look.

navidcy commented:
OK. Few thoughts on @ChrisC28 comment:

The counter-devil's advocate asks:

Why we need higher resolutions tho?

Just making the (minor?) point we all ocean-fluids people take for granted is not apparent to all. I was talking to a cosmologist the other day and I had to explain that. I think it boils down to that fluids have forward energy cascade and that was the motivation before adding these in. Very happy to remove it tho!

navidcy commented:
btw, this sentence

> Regional ocean models also allow for local regions of interest to downstream users that may be contained within a single grid cell of a coarse resolution model to be resolved.

doesn't make sense and I also can't sniff what you might wanted to say here @ChrisC28. Did you just wanna say that we can dig into and simulate the details? (I doubt..)

aekiss commented on paper/paper.md:
```diff
@@ -0,0 +1,124 @@
+
+# Summary
+
+The Modular Ocean Model version 6 (MOM6) is a widely-used open-source general circulation ocean--sea ice model, developed mainly at the NOAA Geophysical Fluid Dynamics Laboratory [@Adcroft2019MOM6].
+MOM6 contains several improvements over its predecessor MOM5 [@griffies2014elements], including the implementation of the Arbitrary-Lagrangian-Eulerian vertical coordinates [@griffies2020ALE], more efficient tracer advection schemes, and state-of-the art parameterizations of sub-grid scale physics.
```
I think it would read better if this was the 3rd paragraph.

aekiss commented on paper/paper.md:
```diff
@@ -0,0 +1,124 @@
+The Modular Ocean Model version 6 (MOM6) is a widely-used open-source general circulation ocean--sea ice model, developed mainly at the NOAA Geophysical Fluid Dynamics Laboratory [@Adcroft2019MOM6].
+MOM6 contains several improvements over its predecessor MOM5 [@griffies2014elements], including the implementation of the Arbitrary-Lagrangian-Eulerian vertical coordinates [@griffies2020ALE], more efficient tracer advection schemes, and state-of-the art parameterizations of sub-grid scale physics.
+
+The nature of turbulent flows is such that smaller scales of motion emerge spontaneously, i.e., flows exhibit forward energy cascade [@richardson1922weather].
```
If this becomes the 1st paragraph, open with a brief summary of what this package does, e.g. "The regional-mom6 Python package provides an easy and versatile way to set up regional configurations of the MOM6 ocean model."

aekiss commented on paper/paper.md:
```diff
@@ -0,0 +1,124 @@
+
+# Summary
+
+The Modular Ocean Model version 6 (MOM6) is a widely-used open-source general circulation ocean--sea ice model, developed mainly at the NOAA Geophysical Fluid Dynamics Laboratory [@Adcroft2019MOM6].
```
```suggestion
The Modular Ocean Model version 6 (MOM6) is a widely-used open-source general circulation ocean--sea ice model, written in Fortran and developed mainly at the NOAA Geophysical Fluid Dynamics Laboratory [@Adcroft2019MOM6].
```

aekiss commented on paper/paper.md:
```diff
@@ -0,0 +1,124 @@
+Despite the increase in computational power and the use of graphical processing units that can bring breakthrough performance and speedup [@silvestri2023oceananigansjl], there will always be processes that are smaller than the model's grid spacing and, thus, remain unresolved.
+
+To resolve more scales of motion within the constraints of computational power, we can turn to regional ocean modeling.
+Regional ocean modeling simulates the ocean only at a prescribed region which is a subset of the global ocean.
```
```suggestion
Regional ocean modeling simulates the ocean only in a prescribed region which is a subset of the global ocean.
```

aekiss commented on paper/paper.md:
```diff
@@ -0,0 +1,124 @@
+
+To resolve more scales of motion within the constraints of computational power, we can turn to regional ocean modeling.
+Regional ocean modeling simulates the ocean only at a prescribed region which is a subset of the global ocean.
+To do that, we need to apply open boundary conditions at the region's boundaries, that is, we need to impose conditions that mimic the oceanic flow that we are not simulating (open boundary conditions; @Orlanski1976).
```
```suggestion
To do that, we need to apply open boundary conditions at the region's boundaries, that is, we need to impose conditions that mimic the oceanic flow that we are not simulating (Orlanski1976).
```

aekiss commented on paper/paper.md:
```diff
@@ -0,0 +1,124 @@
+To resolve more scales of motion within the constraints of computational power, we can turn to regional ocean modeling.
+Regional ocean modeling simulates the ocean only at a prescribed region which is a subset of the global ocean.
+To do that, we need to apply open boundary conditions at the region's boundaries, that is, we need to impose conditions that mimic the oceanic flow that we are not simulating (open boundary conditions; @Orlanski1976).
+\autoref{fig:tasman} shows the surface currents from a regional ocean simulation of the Tasman sea that was configured using the `regional-mom6` package.
```
```suggestion
For example, \autoref{fig:tasman} shows the surface currents from a regional ocean simulation of the Tasman sea that was configured using the `regional-mom6` package.
```

aekiss commented on paper/paper.md:
```diff
@@ -0,0 +1,124 @@
+Regional ocean modeling simulates the ocean only at a prescribed region which is a subset of the global ocean.
+To do that, we need to apply open boundary conditions at the region's boundaries, that is, we need to impose conditions that mimic the oceanic flow that we are not simulating (open boundary conditions; @Orlanski1976).
+\autoref{fig:tasman} shows the surface currents from a regional ocean simulation of the Tasman sea that was configured using the `regional-mom6` package.
+The boundaries of the domain depicted in \autoref{fig:tasman}b are forced with the ocean flow from a reanalysis product that is shown in \autoref{fig:tasman}a.
```
need 2 panels in Fig 1

aekiss replied:
```suggestion
The boundaries of the domain depicted in \autoref{fig:tasman}b are forced with the ocean flow from a global data product product that is shown in \autoref{fig:tasman}a.
```
("reanalysis" is specialist jargon)

aekiss commented on paper/paper.md:
```diff
@@ -0,0 +1,124 @@
+
+![A snapshot of the ocean surface currents from a MOM6 regional simulation of the Tasman sea. The simulation is forced by GLORYS and ERA5 reanalysis datasets and configured with a horizontal resolution of 1/80th degree and 100 vertical levels (see @tasmantides for the source code). \label{fig:tasman}](tasman_speed.png){ width=80% }
+
+The `regional-mom6` package takes as input various datasets that contain the ocean initial condition, the boundary forcing (ocean and atmosphere) for the regional domain, and the bathymetry.
```
```suggestion
The `regional-mom6` package takes as input various datasets that contain the ocean initial condition, the boundary forcing (ocean and atmosphere) for the regional domain, and the seafloor topography.
```
avoid "bathymetry" jargon?

aekiss replied:
Also, would a schematic be useful to illustrate this?

aekiss commented on paper/paper.md:
```diff
@@ -0,0 +1,124 @@
+
+The `regional-mom6` package takes as input various datasets that contain the ocean initial condition, the boundary forcing (ocean and atmosphere) for the regional domain, and the bathymetry.
+The input datasets can be on the Arakawa A, B, or C grids [@arakawa1977computational]; the package performs the appropriate interpolation using `xESMF` [@xesmf] under the hood, to put the everything on the C grid required by MOM6.
+This base grid for the regional configuration can be constructed in two ways.
+The first, by the user defining a desired resolution and choosing between pre-configured options.
+The second, by the user providing a pre-existing horizontal and/or vertical MOM6 grids.
```
```suggestion
This base grid for the regional configuration can be constructed in two ways,
either by the user defining a desired resolution and choosing between pre-configured options,
or by the user providing a pre-existing horizontal and/or vertical MOM6 grids.
```

aekiss commented on paper/paper.md:
```diff
@@ -0,0 +1,124 @@
+This base grid for the regional configuration can be constructed in two ways.
+The first, by the user defining a desired resolution and choosing between pre-configured options.
+The second, by the user providing a pre-existing horizontal and/or vertical MOM6 grids.
+The user can use MOM6's Arbitrary-Lagrangian-Eulerian vertical coordinates, regardless of the native vertical coordinates of the boundary forcing input.
```
```suggestion
The user can use MOM6's Arbitrary-Lagrangian-Eulerian vertical coordinates (Griffies et al., 2020), regardless of the native vertical coordinates of the boundary forcing input.
```

Griffies, S. M., Adcroft, A., and Hallberg, R. W. (2020). A primer on the vertical Lagrangian-remap method in ocean models based on finite volume generalized vertical coordinates. Journal of Advances in Modeling Earth Systems, 12(10). https://doi.org/10.1029%2F2019ms001954

aekiss replied:
not sure if that's the primary reference?

navidcy replied:
Hm.. OK.. @ashjbarnes?

aekiss commented on paper/paper.md:
```diff
@@ -0,0 +1,124 @@
+The second, by the user providing a pre-existing horizontal and/or vertical MOM6 grids.
+The user can use MOM6's Arbitrary-Lagrangian-Eulerian vertical coordinates, regardless of the native vertical coordinates of the boundary forcing input.
+The package automates the re-gridding of all the required forcing input, takes care of all the metadata encoding, generates the regional grid, and ensures that the final input files are in the format expected by MOM6.
+Additionally, the tricky case of a regional configuration that includes the 'seam' in the longitude of the raw input data is handled automatically, removing the need for any preprocessing of the input data.
+(For example, such a 'seam'-related issue arises for a 10ᵒ-wide regional configuration centered at Fiji (178ᵒE) when forced by input with native longitude coordinate in the range between 180ᵒW and 180ᵒE.)
```
```suggestion
Additionally, the tricky case of a regional configuration that includes the 'seam' in the longitude of the raw input data (e.g. a 10ᵒ-wide regional configuration centred at Fiji (178ᵒE) and forced by input with native longitude coordinate in the range 180ᵒW -- 180ᵒE) is handled automatically, removing the need for any preprocessing of the input data.
```

aekiss replied:
Is the Arctic tripole seam also handled?

navidcy replied:
I think no because one limitation of the package is that the regional config boundaries are lines of constant lat/lon. Right @ashjbarnes ?

aekiss commented on paper/paper.md:
```diff
@@ -0,0 +1,124 @@
+The package automates the re-gridding of all the required forcing input, takes care of all the metadata encoding, generates the regional grid, and ensures that the final input files are in the format expected by MOM6.
+Additionally, the tricky case of a regional configuration that includes the 'seam' in the longitude of the raw input data is handled automatically, removing the need for any preprocessing of the input data.
+(For example, such a 'seam'-related issue arises for a 10ᵒ-wide regional configuration centered at Fiji (178ᵒE) when forced by input with native longitude coordinate in the range between 180ᵒW and 180ᵒE.)
+The above-mentioned automation allows users to setup a regional MOM6 configuration using only Python and from the convenience of a single Jupyter notebook.
```
```suggestion
This automation allows users to set up a regional MOM6 configuration using only Python and from the convenience of a single Jupyter notebook.
```

aekiss commented on paper/paper.md:
```diff
@@ -0,0 +1,124 @@
+Additionally, the tricky case of a regional configuration that includes the 'seam' in the longitude of the raw input data is handled automatically, removing the need for any preprocessing of the input data.
+(For example, such a 'seam'-related issue arises for a 10ᵒ-wide regional configuration centered at Fiji (178ᵒE) when forced by input with native longitude coordinate in the range between 180ᵒW and 180ᵒE.)
+The above-mentioned automation allows users to setup a regional MOM6 configuration using only Python and from the convenience of a single Jupyter notebook.
+Rules-of-thumb to guide the user in setting grid parameters such as the regional domains resolution, can be found in the paper by @Herzfeld2011.
```
```suggestion
@Herzfeld2011 provides rules of thumb to guide the user in setting regional grid parameters such as the resolution.
```

aekiss commented on paper/paper.md:
```diff
@@ -0,0 +1,124 @@
+Subsequently, a user need only copy a demo notebook, modify the longitude, latitude and resolution, and simply by running the notebook from start to finish will generate all they need for running a MOM6 experiment in their domain of interest.
+-->
+
+`regional-mom6` is installable via `conda`, it is continuously tested, and comes with an extensive documentation that also includes documented tutorials and examples for setting up regional MOM6 configurations using publicly-available forcing and bathymetry datasets (namely, the GLORYS dataset for ocean boundary forcing [@glorys], the ERA5 reanalysis for atmospheric forcing [@era5], and the GEBCO ...
```
```suggestion
`regional-mom6` is installable via `conda`, it is continuously tested, and comes with extensive documentation including tutorials and examples for setting up regional MOM6 configurations using publicly-available forcing and bathymetry datasets (namely, the GLORYS dataset for ocean boundary forcing [@glorys], the ERA5 reanalysis for atmospheric forcing [@era5], and the GEBCO dataset for seafloor topography [@gebco]).
```

aekiss commented on paper/paper.md:
```diff
@@ -0,0 +1,124 @@
+Besides making regional modelling with MOM6 more accessible, our package can automate the generation of multiple experiments (e.g., a series of perturbation experiments), saving time and effort, and improving reproducibility. 
+
+We designed `regional-mom6` with automation of regional configurations in mind.
+However, the package's code design and modularity makes more complex configurations possible since users can use their own custom-made grids with more complex boundaries and construct the boundary forcing terms one by one.
```
```suggestion
However, the package's code design and modularity make more complex configurations possible since users can use their own custom-made grids with more complex boundaries and construct the boundary forcing terms one by one.
```

aekiss commented on paper/paper.md:
```diff
@@ -0,0 +1,124 @@
+
+The learning curve for setting up a regional ocean model can be quite steep, and it is not obvious for a new user what inputs are required, nor the appropriate format.
+In the case of MOM6, there are several tools scattered in Github repositories, for example those collected in Earth System Modeling Group grid tools [@gridtools].
+Also, there exist several regional configuration examples (e.g., [ADD 1-2 CITATIONS OF REPOS HERE?]) but they are hardcoded for particular domains, specific input files, and work only on specific high-performance computing machines.
```
[ADD 1-2 CITATIONS OF REPOS HERE?]

aekiss commented on paper/paper.md:
```diff
@@ -0,0 +1,124 @@
+Also, there exist several regional configuration examples (e.g., [ADD 1-2 CITATIONS OF REPOS HERE?]) but they are hardcoded for particular domains, specific input files, and work only on specific high-performance computing machines.
+
+Until now there has been no one-stop-shop for users to learn how to get a regional MOM6 configuration up and running.
+Users are required to use several tools in several programming languages and then modify --sometimes by hand-- some of the input metadata to bring everything into the format that MOM6 expects.
```
```suggestion
Users are required to use several tools in several programming languages and then modify -- sometimes by hand -- some of the input metadata to bring everything into the format that MOM6 expects.
```

aekiss commented on paper/paper.md:
```diff
@@ -0,0 +1,124 @@
+
+# Statement of need
+
+The learning curve for setting up a regional ocean model can be quite steep, and it is not obvious for a new user what inputs are required, nor the appropriate format.
```
```suggestion
The learning curve for setting up a regional ocean model can be steep, and it is not obvious for a new user what inputs are required, nor the appropriate format.
```

aekiss reviewed (approved):
Looks good, I've just added a few suggestions that might improve the flow.

aekiss commented on paper/paper.md:
```diff
@@ -0,0 +1,124 @@
+---
+title: 'regional-mom6: Automatic generation of regional configurations for the Modular Ocean Model 6 in Python'
```
```suggestion
title: 'regional-mom6: Automatic generation of regional configurations for the Modular Ocean Model v6 in Python'
```

navidcy commented:
> Looks good, I've just added a few suggestions that might improve the flow.

Thanks @aekiss! I accepted most of your suggestions and we will work on incorporating the rest.

navidcy commented:
> @ashjbarnes suggested that instead of my figure suggestion we show this:
> 
> ![regional](https://private-user-images.githubusercontent.com/7112768/323318333-a308b606-055e-482a-abdb-343785305a7e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3Nzg0MzcsIm5iZiI6MTcxMzc3ODEzNywicGF0aCI6Ii83MTEyNzY4LzMyMzMxODMzMy1hMzA4YjYwNi0wNTVlLTQ4MmEtYWJkYi0zNDM3ODUzMDVhN2UucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MjJUMDkyODU3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTY2ZTE4ZGY4NWNiNjMyOTNlNDY5ZDRhYWIzNzBjNzlhMjBlMzFiNzYzMzg5MDk4ZDljMzNiM2UxMzZiZTZhYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.50rp0xkAVGgLrL57ZGEeIt1ErNKgjCZ0NwnVDHeJiB0)
> 
> This showcases how a regional model matches the flow at its boundaries and how we can use regional modeling to resolve more and more scales of motion.
> 
> @reillyja, you made this figure, right? is it easy to reproduce it with a few tweaks and use it in the paper? could you point me to a notebook that produces this figure perhaps?

@reillyja?

navidcy commented:
@ChrisC28 and @aekiss, I revised the first 3 paragraphs to incorporate your suggestions. Is it better now?

navidcy commented on paper/paper.md:
```diff
@@ -0,0 +1,122 @@
+Regional ocean models can be run at higher resolutions while limiting the required computational resources.
+Higher-resolution regional ocean models improve the representation of smaller-scale motions, such as tidal beams, mixing, mesoscale and sub-mesoscale circulation, as well as the oceanic response to smaller-scale bathymetric or coastal features (such as headlands, islands, sea-mounts, or submarine canyons) and surface forcing (such as atmospheric fronts and convective storms).
+
+MOM6 is a widely-used open-source general circulation ocean--sea ice model, written in Fortran and developed mainly at the NOAA Geophysical Fluid Dynamics Laboratory [@Adcroft2019MOM6].
+MOM6 contains several improvements over its predecessor MOM5 [@griffies2014elements], including the implementation of the Arbitrary-Lagrangian-Eulerian vertical coordinates [bleck2002gvc; @griffies2020ALE], more efficient tracer advection schemes, and state-of-the art parameterizations of sub-grid scale physics.
```
in these two sentences I think we enter a bit too much into describing MOM6...
E.g., why do we need to mention if it's "developed mainly by GFDL" or not? Or why is it relevant whether it has better parametrisations?

@ashjbarnes, @aekiss, @ChrisC28

ChrisC28 commented:
> btw, this sentence
> 
> > Regional ocean models also allow for local regions of interest to downstream users that may be contained within a single grid cell of a coarse resolution model to be resolved.
> 
> doesn't make sense to me and I also can't sniff what you might wanted to say here @ChrisC28. Did you just wanna say that we can dig into and simulate the details? (I doubt..)

Hey @navidcy 
Kinda muddled up my words. To give an example, I work a lot with Marine Park managers and governments in Pacific Island countries. Both want to know what is happening around the island or area that they look after. But in a climate model, their entire island or marine park is contained within one grid box! A high res regional model lets you look at the variation within a grid-box scale area. 
Reword something like this: 
"Regional modelling also allows for the "downscaling" of coarse resolution ocean or climate models, permitting the representation of variation in local conditions that might otherwise be contained within few (or even a single) model grid cell" 
Or something like that....

aekiss commented on paper/paper.md:
```diff
@@ -0,0 +1,119 @@
+`regional-mom6` is a Python package that provides an easy and versatile way to set up regional configurations of the Modular Ocean Model version 6 (MOM6).
+
+In the ocean, fast and small-scale motions (from ~100m to ~100km varying at time scales of hours to days) play an important role in shaping the large-scale ocean circulation and climate (length scales ~10,000km varying at decadal time scales) [@Melet2022ch2; @deLavergne2022ch3; @Gula2022ch8].
+Despite the increase in computational power and the use of graphical processing units that can bring breakthrough performance and speedup [@silvestri2023oceananigansjl], there are always processes, boundary, or forcing features that are smaller than the model's grid spacing and, thus, remain unresolved in operational ocean models.
```
```suggestion
Despite the increase in computational power and the use of graphical processing units that can bring breakthrough performance and speedup [@silvestri2023oceananigansjl], there are always processes, boundary, or forcing features that are smaller than the model's grid spacing and, thus, remain unresolved in global ocean models.
```
"operational" means something different (ocean forecasting systems)

aekiss commented:
> @ChrisC28 and @aekiss, I revised the first 3 paragraphs to incorporate your suggestions.
> 
> * cited Bleck 2002 regarding ALE -- not totally sure if that's the right ref
> * deleted sentence about energy cascades
> * added first summary-sentence for what package does, moved discussion about MOM6 bit further down
> * used Chris' motivation for "why regional ocean modeling"... There was [a sentence I couldn't decipher](https://github.com/COSIMA/regional-mom6/pull/100#issuecomment-2066351008) and I dropped it; @ChrisC28 feel free to put it back but rephrase it perhaps because at its current form I couldn't understand what you wanted to say
> 
> Is it better now?

yep! I've added some minor suggestions.

Bleck 2002 is a good primary reference

navidcy commented on paper/paper.md:
```diff
@@ -0,0 +1,119 @@
+  - name: Chris Chapman
+    orcid: 0000-0002-6030-1951
+    affiliation: 5
+  - name: John Reilly
+    affiliation: 6
```
@reillyja do you have an Orcid number? if so, can you add it?

ashjbarnes commented:
Repos to cite in line 87: https://github.com/jsimkins2/nwa25

navidcy commented:
hi team, the package is submitted to JOSS, you can watch [this issue](https://github.com/openjournals/joss-reviews/issues/6686) for pre-review

navidcy commented:
hi team,

me and @ashjbarnes came up with a list of suggested reviewers; see https://github.com/openjournals/joss-reviews/issues/6686#issuecomment-2080380181

just checking the nobody of you is conflicted with them (coauthored a paper or are on the same grant)

cc @aekiss, @ChrisC28

reillyja commented:
> @ashjbarnes suggested that instead of my figure suggestion we show this:
> 
> ![regional](https://private-user-images.githubusercontent.com/7112768/323318333-a308b606-055e-482a-abdb-343785305a7e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQzNDU4MTUsIm5iZiI6MTcxNDM0NTUxNSwicGF0aCI6Ii83MTEyNzY4LzMyMzMxODMzMy1hMzA4YjYwNi0wNTVlLTQ4MmEtYWJkYi0zNDM3ODUzMDVhN2UucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MjhUMjMwNTE1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDM3YmZkMjU0ODA4YjA2OWNiZjNjYjdkOTcwODcxZWI3MWYzOTA1NzRlNDM5YjY4NTdiM2UwNzU4ZjI5NzgyNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.-hcsNcKfijIHd9t5Tq3ZeVS3LksFI62Z0HuVmASno8U)
> 
> This showcases how a regional model matches the flow at its boundaries and how we can use regional modeling to resolve more and more scales of motion.
> 
> @reillyja, you made this figure, right? is it easy to reproduce it with a few tweaks and use it in the paper? could you point me to a notebook that produces this figure perhaps?

Hey guys, Navid just pointed me to this. I've simplified the notebook to [recreate that figure here](https://github.com/reillyja/phd_notebooks/tree/master/for_joss_paper) Somehow I haven't been able to find the exact timestep that showcases the nice eddy train in the eac extension region but I can try to find this timestep when I have more time if that is preferred. 

Happy to either give access to the model outputs in my scratch directory, or move the u/v files somewhere more accessible if others want to play around with the plots.

Also happy to take any instruction on how to improve this and I'll work on it!

navidcy commented:
@reillyja I can't run the notebook you posted. Can you ensure that when you restart the kernel all cell can be evaluated sequentially?

I get the following:


```python
year_month = '1992_12'
# access velocities
om2_u_path = glob.glob(access + f'output*/ocean/*u*{year_month}*.nc')
om2_v_path = glob.glob(access + f'output*/ocean/*v*{year_month}*.nc')
om2_u = xr.open_dataset(om2_u_path[0])['u']
om2_v = xr.open_dataset(om2_v_path[0])['v']

om2_vel = xr.merge([om2_u, om2_v])
om2_vel
```

```python
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
Cell In[6], line 5
      3 om2_u_path = glob.glob(access + f'output*/ocean/*u*{year_month}*.nc')
      4 om2_v_path = glob.glob(access + f'output*/ocean/*v*{year_month}*.nc')
----> 5 om2_u = xr.open_dataset(om2_u_path[0])['u']
      6 om2_v = xr.open_dataset(om2_v_path[0])['v']
      8 om2_vel = xr.merge([om2_u, om2_v])

IndexError: list index out of range
```

navidcy commented:
oh... I think `access = '/g/data/ik11/outputs/access-om2-01/01deg_jra55v150_iaf_cycle1/'` fixed it

navidcy commented:
@reillyja I update the figure.

Could you update the caption to describe your figure?

navidcy commented:
congrats all!
paper [is published](https://github.com/openjournals/joss-reviews/issues/6857#issuecomment-2278409887)

![c00a5fe7-b0de-5bfa-5c31-e97085fe906f](https://github.com/user-attachments/assets/69728e94-2b69-4f3b-8c7a-c89a3a5fd3cf)

@PaulSpence (left) handing the 2024 COSIMA Most Selfless Contributor Award to @ashjbarnes (right); July 1st, 2024, Canberra, Australia
